### PR TITLE
feat(server): add list_files WS endpoint with recursive walk and gitignore (#1285)

### LINKED ISSUE #1285: feat(dashboard): add server-side file listing WS endpoint
## Summary

Add a WebSocket message handler that lists files from the session's working directory, respecting .gitignore patterns.

Part of #1282

## Implementation Plan

- Files to modify: `packages/server/src/ws-message-handlers.js`, `packages/server/src/ws-schemas.js`
- Add `list_files` client→server message type with optional `query` param for filtering
- Handler reads CWD, walks directory tree (max depth ~3), filters via .gitignore
- Returns `file_list` server→client message with array of `{ path, type, size }`
- Add tests in `packages/server/tests/`

## Acceptance Criteria

- [ ] `list_files` message type added to WS schema
- [ ] Handler returns files from session CWD
- [ ] Respects .gitignore patterns
- [ ] Optional `query` param filters results (substring match)
- [ ] Max depth limit prevents excessive traversal
- [ ] Tests cover happy path and edge cases

### PR DESCRIPTION
**Pull request by blamechris**
## Summary

- Add `list_files` client→server WS message type for recursive file listing
- Walk session CWD up to 3 levels deep, respecting .gitignore patterns
- Skip hidden files/dirs and node_modules automatically
- Optional `query` param for case-insensitive substring filtering
- Returns `file_list` server→client with `{ path, type, size }` entries
- Wired into both multi-session and single-session message handlers

Closes #1285

## Test Plan

- [x] 16 unit tests pass (traversal, depth, gitignore, query, metadata, errors)
- [x] 2 schema validation tests pass
- [x] Existing server tests unbroken
- [ ] Manual: send `{ type: "list_files" }` over WS, verify file list returned

### CODE REVIEW

File: packages/server/src/ws-file-ops.js [hunk 1]
```diff
@@ -592,12 +592,166 @@ export function createFileOps(sendFn) {
...
+        for (const d of dirents) {
+          if (files.length >= MAX_FILES) break
+          if (d.name.startsWith('.')) continue
+          if (d.name === 'node_modules') continue
+
+          const absPath = join(dir, d.name)
+          const relPath = relative(cwdReal, absPath)
+
+          // Check gitignore
+          if (isIgnored(relPath, gitignoreRules)) continue
```
[the remark this answers is not in the document]
Comment (blamechris) [reply packages/server/src/ws-file-ops.js]: **FALSE POSITIVE**

**Reason:** Chroxy server runs exclusively on macOS/Linux per CLAUDE.md. `path.relative()` returns POSIX `/` separators on these platforms. Windows is not a supported target.

**Evidence:**
- CLAUDE.md specifies the server runs on dev machines, macOS primary
- All existing file ops use `relative()` and `join()` without Windows normalization
- The server is designed to run behind Cloudflare tunnels on Unix-like systems
